src/DynamoCore/Models/CustomNodeWorkspaceModel.cs

```diff
@@ -196,20 +196,23 @@ protected virtual void OnFunctionIdChanged(Guid oldId)
 
         public override bool SaveAs(string newPath, ProtoCore.RuntimeCore runtimeCore, bool isBackUp = false)
         {
-            var originalPath = FileName;
-
-            // A SaveAs to an existing function id prompts the creation of a new 
-            // custom node with a new function id
-            if (originalPath != newPath)
+            if (!isBackUp)
             {
-                FileName = newPath;
-                // If it is a newly created node, no need to generate a new guid
-                if (!string.IsNullOrEmpty(originalPath))
-                    CustomNodeId = Guid.NewGuid();
-
-                // This comes after updating the Id, as if to associate the new name
-                // with the new Id.
-                SetInfo(Path.GetFileNameWithoutExtension(newPath));
+                var originalPath = FileName;
+
+                // A SaveAs to an existing function id prompts the creation of a new 
+                // custom node with a new function id
+                if (originalPath != newPath)
+                {
+                    FileName = newPath;
+                    // If it is a newly created node, no need to generate a new guid
+                    if (!string.IsNullOrEmpty(originalPath))
+                        CustomNodeId = Guid.NewGuid();
+
+                    // This comes after updating the Id, as if to associate the new name
+                    // with the new Id.
+                    SetInfo(Path.GetFileNameWithoutExtension(newPath));
+                }
             }
 
             return base.SaveAs(newPath, runtimeCore);
```

Comment:
It depends on whether we need to add the backed up files to the recent file list. 

I will adjust it to following the older implementation.